File: src/main/java/com/internship/tmontica/dto/Menu.java
package com.internship.tmontica.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.apache.ibatis.type.Alias;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;
import java.util.Date;
import java.util.ArrayList;
import java.util.List;


@Alias("menu")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class  Menu {
    private int id;
    @NotEmpty
    private String nameEng;
    @Min(0)
    private int productPrice;
    @NotEmpty
    private String categoryKo;
    @NotEmpty
    private String categoryEng;
    private boolean monthlyMenu;
    private boolean usable;
    @NotEmpty
    private String imgUrl;
    private String description;
    @Min(0)
    private int sellPrice;
    @Min(0)
    private int discountRate;
    private Date createdDate;
    private Date updatedDate;
    private String creatorId;
    private String updaterId;

    @Min(0)
    private int stock;
    @NotEmpty
    private String nameKo;

    // 메뉴의 옵션
//    private List<Option> options;
//
//    public void addOption(Option option){
//        if(options == null)
//            options = new ArrayList<>();
//        options.add(option);
//    }

}

File: front-end/src/types/menu.ts
export type TMenusItem = {
  id: number;
  nameKo: string;
  nameEng: string;
  imgUrl: string;
  stock: number;
};

export interface TBasicMenuOption {
  id: number;
  quantity: number;
}

export interface TMenuOption extends TBasicMenuOption {
  type: string;
  name: string;
  price: number | 0;
}

export interface TMenu {
  id: number;
  nameEng: string;
  nameKo: string;
  description: string;
  imgUrl: string;
  sellPrice: number;
  discountRate: number;
  category: string;
  stock: number;
  monthlyMenu: boolean;
  option: Array<TMenuOption>;
  getOptionById(id: number): TMenuOption;
}

export interface TOrderDetail {
  id: number;
  nameEng: string;
  nameKo: string;
  description: string;
  imgUrl: string;
  sellPrice: number;
  discountRate: number;
  category: string;
  stock: number;
  monthlyMenu: boolean;
  option: string;
  getOptionById(id: number): TMenuOption;
  quantity: number;
}

export type TBasicMenuOptionArray = Array<TBasicMenuOption>;
export type TMenuOptionMap = Map<string, TMenuOption>;
